### PR TITLE
xmonadctl: init at 0.17.0

### PR DESCRIPTION
**Pull request by axgfn**
###### Description of changes

Add `xmonadctl`, a simple script to send commands to a running instance of xmonad. (Note that XMonad must first be configured to listen for commands using the `XMonad.Hooks.ServerMode` module.)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/window-managers/xmonad/xmonadctl.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  version = "0.17.0";
+
+  src = fetchFromGitHub {
+    owner = "xmonad";
+    repo = "xmonad-contrib";
+    rev = "v${version}";
+    sha256 = "142ycg7dammq98drimv6xbih8dla9qindxds9fgkspmrrils3sar";
+  };
+
+  installPhase = ''
```
Comment (viraptor) [inline pkgs/applications/window-managers/xmonad/xmonadctl.nix:15]:
```suggestion
  installPhase = ''
    runHook preInstall
```

File: pkgs/applications/window-managers/xmonad/xmonadctl.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    mkdir -p $out/bin
+    ${xmonadctlEnv}/bin/ghc -o $out/bin/xmonadctl \
+      --make scripts/xmonadctl.hs
+  '';
```
Comment (viraptor) [inline pkgs/applications/window-managers/xmonad/xmonadctl.nix:19]:
```suggestion
    runHook postInstall
  '';
```

**Comment by viraptor** [comment]
Result of `nixpkgs-review pr 181163` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xmonadctl</li>
  </ul>
</details>

File: pkgs/applications/window-managers/xmonad/xmonadctl.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
```
Comment (sternenseemann) [inline pkgs/applications/window-managers/xmonad/xmonadctl.nix:21]:
```suggestion
  meta = with lib; {
    platforms = platforms.unix;
```

**Review by sternenseemann — COMMENTED** [review]
Can you add your package to `pkgs/top-level/release-haskell.nix` to the section with the top level haskell packages? This will make sure CI tests this package & failures are noticed.

File: pkgs/applications/window-managers/xmonad/xmonadctl.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchFromGitHub, ghcWithPackages, ... }:
+
+let xmonadctlEnv = ghcWithPackages (self: [ self.xmonad-contrib self.X11 ]);
+in stdenv.mkDerivation rec {
+  pname = "xmonadctl";
+  version = "0.17.0";
+
+  src = fetchFromGitHub {
```
[the remark this answers is not in the document]
Comment (ajgrf) [reply pkgs/applications/window-managers/xmonad/xmonadctl.nix:8]: Sure, I'll open up an issue for this. I would note, though, that this script is extremely simple and will likely continue to work fine even if it's not updated in sync with xmonad-contrib.

[the remark this answers is not in the document]
Comment (ajgrf) [reply pkgs/applications/window-managers/xmonad/xmonadctl.nix:8]: This is now already done upstream and will be included in the next release. With any luck, `xmonadctl` will only ever need to be updated once (for the 0.18.0 release or whatever the next version number is).

[the remark this answers is not in the document]
Comment (sternenseemann) [reply pkgs/applications/window-managers/xmonad/xmonadctl.nix:8]: Great!

**Pull request merged**